models/dataset_test.go

```diff
@@ -190,44 +190,47 @@ func TestCreateDataset(t *testing.T) {
 }
 
 func TestCreateVersion(t *testing.T) {
-	t.Parallel()
-	Convey("Successfully return without any errors", t, func() {
-		Convey("when the version has all fields", func() {
-			b, err := json.Marshal(associatedVersion)
+		t.Parallel()
+		Convey("Successfully return without any errors", t, func() {
+			Convey("when the version has all fields", func() {
+				testDatasetID := "test-dataset-id"
+				b, err := json.Marshal(associatedVersion)
+				if err != nil {
+					t.Logf("failed to marshal test data into bytes, error: %v", err)
+					t.FailNow()
+				}
+				r := bytes.NewReader(b)
+				version, err := CreateVersion(r, testDatasetID)
+				So(err, ShouldBeNil)
+				So(version.CollectionID, ShouldEqual, collectionID)
+				So(version.Dimensions, ShouldResemble, []Dimension{dimension})
+				So(version.Downloads, ShouldResemble, &downloads)
+				So(version.Edition, ShouldEqual, "2017")
+				So(version.ID, ShouldNotBeNil)
+				So(version.ReleaseDate, ShouldEqual, "2017-10-12")
+				So(version.LatestChanges, ShouldResemble, &[]LatestChange{latestChange})
+				So(version.Links.Spatial.HRef, ShouldEqual, "http://ons.gov.uk/geographylist")
+				So(version.State, ShouldEqual, AssociatedState)
+				So(version.Temporal, ShouldResemble, &[]TemporalFrequency{temporal})
+				So(version.Version, ShouldEqual, 1)
+				So(version.DatasetID, ShouldEqual, testDatasetID)
```

Comment:
Think about ordering the assertions in alphabetical order of the field you are checking, easier to see if the field you are interested is in the list of assertions.